tests/lib/uuid/src/main.c

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright (c) 2024, SECO Mind Srl
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/ztest.h>
+#include <zephyr/sys/uuid.h>
+
+ZTEST(uuid, test_uuid_v4)
+{
+	uuid_t uuid_1 = {0};
+	uuid_t uuid_2 = {0};
+
+	uuid_generate_v4(uuid_1);
+	/* Check version and variant fields */
+	zassert_equal(uuid_1[6U] >> 4U, 4U, "Generated UUID v4 contains an incorrect 'ver' field");
+	zassert_equal(uuid_1[8U] >> 6U, 2U, "Generated UUID v4 contains an incorrect 'var' field");
+	/* Check another generated UUID is different */
+	uuid_generate_v4(uuid_2);
+	zassert(memcmp(uuid_1, uuid_2, UUID_SIZE) != 0, "The two generated UUID v4 are equal.");
```

Comment:
This is a bit of a funny test - 2 generated UUIDs can be the same, although very unlikely. We can probably keep it, but it actually fail with correct behavior